Scripts/DowloadPortalItemData.swift

```diff
@@ -245,31 +246,46 @@ var downloadedItems = previousDownloadedItems
 let dispatchGroup = DispatchGroup()
 
 portalItems.forEach { portalItem in
+    // Checks to see if an item is already downloaded.
+    guard !downloadedItems.keys.contains(portalItem.identifier) else {
+        print("note: Item already downloaded: \(portalItem.identifier)")
+        return
+    }
+    
     let destinationURL = downloadDirectoryURL.appendingPathComponent(portalItem.identifier, isDirectory: true)
-    // Checks if a directory exists or not, to see if an item is already downloaded.
+    
+    // Deletes the directory if it already exists.
+    // This happens when the item is not in the plist and needs to be redownloaded.
     if FileManager.default.fileExists(atPath: destinationURL.path) {
```

Comment:
From [the doc](https://developer.apple.com/documentation/foundation/filemanager/1415645-fileexists):

> Attempting to predicate behavior based on the current state of the file system or a particular file on the file system is not recommended. Doing so can cause odd behavior or race conditions. It’s far better to attempt an operation (such as loading a file or creating a directory), check for errors, and handle those errors gracefully than it is to try to figure out ahead of time whether the operation will succeed.

Just try deleting the item and don't worry about the error. It is safe to assume that it will only throw an error if the directory doesn't exist, which is perfectly fine:

```swift
try? FileManager.default.removeItem(at: destinationURL)
```